lib/pbench/test/unit/server/test_shell_cli.py

```diff
@@ -309,58 +309,109 @@ def init_indexing(*args, **kwargs) -> int:
         assert ret_val == 1
 
     @staticmethod
-    @pytest.mark.parametrize("init_db_exc", ["section", "option"])
+    @pytest.mark.parametrize(
+        "init_db_exc",
+        [NoSectionError("missingsection"), NoOptionError("section", "missingoption")],
+    )
     def test_main_initdb_failed(
         monkeypatch, make_logger, mock_get_server_config, init_db_exc
     ):
+        def immediate_success(*args, **kwargs):
+            pass
+
+        called = [False]
+
         def init_db(*args, **kwargs) -> int:
-            if init_db_exc == "section":
-                exc = NoSectionError("missingsection")
-            elif init_db_exc == "option":
-                exc = NoOptionError("section", "missingoption")
-            else:
-                exc = Exception(f"Bad test parameter, {init_db_exc}")
-            raise exc
+            called[0] = True
+            raise init_db_exc
 
         monkeypatch.setattr(shell.site, "ENABLE_USER_SITE", False)
+        monkeypatch.setattr(shell, "wait_for_uri", immediate_success)
         monkeypatch.setattr(shell, "init_db", init_db)
 
         ret_val = shell.main()
```

Comment:
Should we be mocking `Database.create_if_missing()`?

Reply:
I don't think so, we are using sqlite for the database.